tests/locale/keymap_or_locale.pm

```diff
@@ -0,0 +1,93 @@
+# SUSE's openQA tests
+#
+# Copyright © 2016-2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Keyboard layout test in console and gdm after boot
+# Maintainer: Martin Loviska <mloviska@suse.com>
+
+use base "opensusebasetest";
+use strict;
+use testapi;
+use utils;
+
+sub verify_default_keymap_textmode {
+    my ($test_string, $tag, $tty, $console) = @_;
+
+    send_key($tty)           if defined($tty);
```

Comment:
I don't really get why we need this, could you please explain? I believe select_console should do the trick, so we don't really need both tty and console variables to be defined. What's about parsing a hash, so method could be called like this:
```perl
($keystrokes, "${expected}_keymap_logged", console => 'user-console');
```

Reply:
Basically there are 3 different test cases, to test the keymap in login prompt, logged as root and logged as ordinary user. I found that easier than to perform the keymap test in prompt, remove typed string, log as root, perform keymap test, ensure serial def permissions for user bernhard and logoff. Afterwards log as ordinary user and repeat the same test. Also it has not been decided yet, whether this particular test should continue with other locale settings test or this part is going to be tested in another openQA job. Of course, it can be adjusted as you propose.

Reply:
I just think that for the first scenario we don't need to switch tty at all (maybe removing would be nice though, if we will run more tests after this one), and for other two select_console does the trick, so we don't need send_key($tty).

Reply:
Indeed! Switching ttys is really redundant in this case. 
Verification:
[locale_minimalx@64bit](http://dhcp228.suse.cz/tests/575) 
[locale_textmode@64bit](http://dhcp228.suse.cz/tests/573)
[locale_gnome@64bit](http://dhcp228.suse.cz/tests/574)

Thanks for hint!